content/tokio/topics/tracing.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Getting started with Tracing
+---
+
+The [`tracing`] crate is a framework for instrumenting Rust programs to collect 
+structured, event-based diagnostic information.
+
+In asynchronous systems like Tokio, interpreting traditional log messages can 
+often be quite challenging. Since individual tasks are multiplexed on the same 
+thread, associated events and log lines are intermixed making it difficult to 
+trace the logic flow. `tracing` expands upon logging-style diagnostics by 
+allowing libraries and applications to record structured events with additional 
+information about *temporality* and *causality* — unlike a log message, a span 
+in `tracing` has a beginning and end time, may be entered and exited by the 
+flow of execution, and may exist within a nested tree of similar spans. In 
+addition, `tracing` spans are *structured*, with the ability to record typed 
+data as well as textual messages.
+
+You can use `tracing` to:
+- observe your application with Open Telemetry-supporting services
+- debug your application with Tokio Console
+- emit logs to `stdout` or `journald`
+- profile where your application is spending time
+
+[`tracing`]: https://docs.rs/tracing
+[`tracing-subscriber`]: https://docs.rs/tracing-subscriber
+
+# Setup
+
+To begin, add [`tracing`] and [`tracing-subscriber`] as dependencies:
+
+```toml
+[dependencies]
+tracing = "0.1.34"
+tracing-subscriber = "0.3"
+```
+
+The [`tracing`] crate provides the API we will use to emit traces. The 
+[`tracing-subscriber`] crate provides some basic utilities for forwarding those 
+traces to external listeners (e.g., `stdout`).
+
+# Subscribing to Traces
```

Comment:
I order the sections in this way because adding instrumentation is a task that can be tackled incrementally. If you install the subscriber first, you'll see the fruits of each addition of instrumentation.

Reply:
okay, that's fine; i don't have a strong opinion about this.